### PR TITLE
implementing getopt() and adding tv_sec flag

### PR DESCRIPTION
**Pull request by prajwaltr93**
This PR implements getopt() to parse arguments,
and also added new flag -t to set tv_sec value.
i have also updated help message and README.md file.
code style guidelines are followed.
@tenox7 hope this helps !

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ requires `libudev-dev`
 usbmon [-n][-c]
   -n do not monitor events
   -c collectd exec plugin mode
+  -t=TV_SEC set time interval for monitoring default:10
```
Comment (tenox7) [inline README.md]: please do not put TV_SEC like this, make it to a "seconds" or something human can read

File: usbmon.c [hunk 2]
```diff
@@ -34,29 +34,30 @@ struct collectd {
...
-    return 0;
-}
-
 enum { INFO, WARNING, ERROR };
 
+void printhelp(void) {
+  printf("usbmon [-h|--help][-n][-c][-t tv_sec] \n\t-h|--help (optional) help\n" \
+    "\t-n        (optional) do not monitor events\n" \
+    "\t-c        (optional) collectd exec plugin mode\n"\
+    "\t-t=TV_SEC (optional) set time interval for monitoring\n");
```
Comment (tenox7) [inline usbmon.c]: same please replace TV_SEC with seconds, also print default value

File: usbmon.c [hunk 3]
```diff
@@ -34,29 +34,30 @@ struct collectd {
...
 
-int isopt(int argc, char **argv, const char *opt) {
-    for(int i = 1; i < argc && strcmp(argv[i], opt) == 0; ++i)
-        return 1;	//opt option has been inputted.
-    return 0;
-}
-
 enum { INFO, WARNING, ERROR };
 
+void printhelp(void) {
```
Comment (tenox7) [inline usbmon.c]: I would prefer usage()

File: usbmon.c [hunk 4]
```diff
@@ -89,32 +90,46 @@ int main(int argc, char **argv) {
     struct collectd cv;
     struct utsname u;
     const char *path, *usbpath, *vendor, *serial, *speed, *action;
-    int fd, ret, co = 0;
+    int fd, ret, co = 0, nomon = 0, opt, tv_sec = 10;
```
Comment (tenox7) [inline usbmon.c]: also here do not call it tv_sec rather, interval

File: usbmon.c [hunk 5]
```diff
@@ -89,32 +90,46 @@ int main(int argc, char **argv) {
...
+        case 'c':
+          co = 1;
+          break;
+
+        case 'h':
+          printhelp();
+          return 0;
+
+        case 't':
+          tv_sec = atoi(optarg);
```
Comment (tenox7) [inline usbmon.c]: this should be interval =

File: usbmon.c [hunk 6]
```diff
@@ -160,20 +175,20 @@ int main(int argc, char **argv) {
     while(1) {
         FD_ZERO(&fds);
         FD_SET(fd, &fds);
-        ti.tv_sec = 10; // TODO(tenox): make this in to a flag
+        ti.tv_sec = tv_sec; // TODO(tenox): make this in to a flag (done)
```
Comment (tenox7) [inline usbmon.c]: remove todo comment instead of adding done

**Review by tenox7 — CHANGES_REQUESTED** [review]
In addition please make check if -n and -c are not specified together, if they are please error out with explanation.

Also in function putval now that we have real interval variable you need to add printing interval like in plain text protocol spec. See https://collectd.org/wiki/index.php/Plain_text_protocol#PUTVAL for refernce.

Thank you!

File: README.md (hunk 1 above)
Comment (prajwaltr93) [reply README.md]: i can change it to -t=SECS, since most unix commands have argument to option in UPPERCASE.

File: usbmon.c (hunk 2 above)
Comment (prajwaltr93) [reply usbmon.c]: i will.

File: usbmon.c (hunk 3 above)
Comment (prajwaltr93) [reply usbmon.c]: that would be appropriate

File: usbmon.c (hunk 4 above)
Comment (prajwaltr93) [reply usbmon.c]: changed it .

File: usbmon.c (hunk 5 above)
Comment (prajwaltr93) [reply usbmon.c]: done .

File: usbmon.c (hunk 6 above)
Comment (prajwaltr93) [reply usbmon.c]: removed it

**Comment by prajwaltr93** [comment]
@tenox7 minor changes are updated.
to enforce mutual exclusivity to options -n and -c is not possible through getopt() itself, thus i will just use simple **if** to check and print appropriate message . this is done in next commit .

**Comment by prajwaltr93** [comment]
@tenox7 small doubt regarding using interval in putval(),
interval variable in putval() can be implemented in following ways :
1.make interval global 
2.modify struct collectd
3.pass it to pulval()
modifying in any above ways will affect further modifications to program, which would be better ?
and will adding interval to PUTVAL complete TODO line 219 output throttling ?

**Comment by prajwaltr93** [comment]
@tenox7 few doubts
1. todo line 148 , can we just reuse logmsg function ?
2. todo line 201 , it already has a different print function , what does it mean?
3. todo line 219, i could not understand what output throttling means .

**Comment by prajwaltr93** [comment]
@tenox7 if both -n and -c are used, main prints usage() and returns.

**Pull request closed without merging**